coast/CLIMATOLOGY.py

```diff
@@ -76,10 +84,122 @@ def make_climatology(
                 ds = ds.drop_vars("clim_mean_ones_tmp")
 
         if fn_out is not None:
-            print("Saving to file. May take some time..")
+            info("Saving to file. May take some time..")
             with ProgressBar():
                 ds_mean.to_netcdf(fn_out)
 
         return ds_mean
 
-        return
+    @staticmethod
+    def _get_date_ranges(years, month_periods):
+        """Calculates a list of datetime date ranges for a given list of years and a specified start/end month.
+
+        Args:
+            years (list): A list of years to calculate date ranges for.
+            month_periods (list): A list containing tuples of start and end month integers.
+            (i.e. [(3,5),(12, 2)] is Mar -> May, Dec -> Feb). Must be in chronological order.
+
+            Returns:
+                date_ranges (list): A list of tuples, each containing a start and end datetime.date object.
+        """
+        date_ranges = []
+        for y in set(years):
+            y = int(y)
+            for period in month_periods:
+                start = period[0]
+                end = period[1]
+                if start > end:
+                    begin_date = date(y, start, 1)
+                    end_day = calendar.monthrange(y + 1, end)[1]
+                    end_date = date(y + 1, end, end_day)
+                else:
+                    begin_date = date(y, start, 1)
+                    end_day = calendar.monthrange(y, end)[1]
+                    end_date = date(y, end, end_day)
+                date_ranges.append((begin_date, end_date))
+        return date_ranges
+
+    @staticmethod
```

Comment:
I think this should probably be a classmethod rather than a staticmethod. Then if something inherits from CLIMATOLOGY it will continue to use the correct class rather than always falling back to this.

Reply:
now a classmethod :)